File: android/src/main/kotlin/com/dns_technologies/mlkit_scanner/commands/StartScanCommand.kt
package com.dns_technologies.mlkit_scanner.commands

import com.dns_technologies.mlkit_scanner.PluginConstants
import com.dns_technologies.mlkit_scanner.commands.base.ScannerCommand
import com.dns_technologies.mlkit_scanner.scanner.Scanner
import com.dns_technologies.mlkit_scanner.utils.requireMap
import com.dns_technologies.mlkit_scanner.utils.requireInt
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.Result

/** Applies a point control to the scanner selected by Dart. */
internal class StartScanCommand(
    scannerProvider: () -> Scanner?,
) : ScannerCommand(scannerProvider) {
    override fun executeCommand(call: MethodCall, result: Result) {
        val delay = call.arguments.requireMap().requireInt(PluginConstants.delayArgument)
        scanner()?.startScan(delay)
        success(result)
    }
}

File: ios/Classes/Commands/StartScanCommand.swift
import Flutter

/// Starts barcode delivery for the selected scanner.
final class StartScanCommand: ScannerCommand {

    /// Validates recognition options and starts barcode delivery.
    override func executeCommand(_ call: FlutterMethodCall, result: @escaping FlutterResult) throws {
        let options = try ScannerMethodArguments.scanOptions(call.arguments)
        try scannerDevice.startScan(type: options.type, delay: options.delay)
        success(result)
    }
}
